src/test/java/de/tum/in/www1/artemis/service/LectureServiceTest.java

```diff
@@ -66,6 +68,14 @@ void initTestCase() throws Exception {
         // always use the lecture and course with the smallest ID, otherwise tests below related to search might fail (in a flaky way)
         course = courseRepository.findByIdWithLecturesAndLectureUnitsElseThrow(courses.stream().min(Comparator.comparingLong(DomainObject::getId)).orElseThrow().getId());
         lecture = course.getLectures().stream().min(Comparator.comparing(Lecture::getId)).orElseThrow();
+        hiddenLecture = course.getLectures().stream().max(Comparator.comparing(Lecture::getId)).orElseThrow();
```

Comment:
lecture and hidden lecture now refer to the same object, this does not seem right.

Reply:
They do not (I am using min and max here and there are two lectures). I didn't want to remove the min/max due to the comment regarding flakiness above. I added an assertion below to ensure that lecture and hiddenLecture are different.

Reply:
oh, did not see that you used min and max.

Can you move this into a local variable of the test case? This lecture is currently only needed in this one test and not somewhere else.

Reply:
Already done ;)